resources/roles/preparation/tasks/rhel.yml

```diff
@@ -0,0 +1,128 @@
+---
+
+- name: Initial fact set
+  set_fact:
+    centos_7_x32: false
+    bats_package: bats
+
+- name: Set fact about CentOS 7 x32 arch
+  set_fact:
+    centos_7_x32: true
+    bats_package: "http://ftp.icm.edu.pl/pub/Linux/dist/epel/7/x86_64/Packages/b/bats-0.4.0-1.20141016git3b33a5a.el7.noarch.rpm"
+  when: >
+    ansible_facts.distribution_major_version == '7' and
+    ansible_facts.userspace_architecture in ('i386', 'i486', 'i586', 'i686')
+
+- name: Re-define procps package name
+  set_fact:
+    procps_pkg: "procps-ng"
+  when: "ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version > '6'"
+
+- set_fact:
+    additional_packages:
+      - "{{ bats_package }}"
+      - file
+      - iproute
+      - "{{ procps_pkg }}"
+
+- set_fact:
+    pip_packages:
+      - python3-pip
+      - python3-setuptools
+  when: "ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version >= '8'"
+
+- set_fact:
+    pip_packages:
+      - python-pip
+      - python-setuptools
+  when: "ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version < '8'"
+
+- name: Add os and updates repositories for version 6
+  yum_repository:
+    name: "centos-6-{{ item }}"
+    baseurl: "{{ centos_repo_baseurl }}/6.10/{{ item }}/$basearch/"
+    description: "CentOS 6 {{ item }} repository"
+    enabled: yes
+    gpgcheck: yes
+    gpgkey: https://vault.centos.org/6.10/os/$basearch/RPM-GPG-KEY-CentOS-6
+  with_items:
+    - "os"
+    - "updates"
+  when: "ansible_distribution_file_variety == 'RedHat' and ansible_facts.distribution_major_version == '6'"
+
+- name: Add YUM proxy
+  ini_file:
+    path: /etc/yum.conf
+    section: main
+    option: proxy
+    value: "{{ package_proxy }}"
+  when: "package_proxy is defined and package_proxy and package_proxy | lower != 'none'"
+
+- name: Remove nodocs param from YUM config
+  ini_file:
+    path: /etc/yum.conf
+    section: main
+    option: tsflags
+    state: absent
+
+- name: Delete /etc/rpm/macros.image-language-conf
+  file:
+    path: /etc/rpm/macros.image-language-conf
+    state: absent
+  when: pytest_is_needed | bool
+
+- name: Install epel-release (8 and higher)
+  yum:
+    name: epel-release
+    state: present
+  when: "ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version >= '8'"
+
+- name: Install epel-release (7 and less)
+  yum:
+    name: "{{ epel_release_url }}"
+    state: present
+  when: > 
+    epel_release_url is defined and
+    epel_release_url and
+    ansible_distribution_file_variety in ('RedHat', 'OracleLinux') and
```

Comment:
Isn't this `ansible_distribution_file_varierty`  condition already evaluated in the role's main task -> https://github.com/AlmaLinux/alts/pull/78/files#diff-2943968aeb4ee6cfceb693b6f6dc47b5313c3ff1d561d4badbdbddcbab0ea861R3-R6?